api/lib/opentelemetry/trace/tracer.rb

```diff
@@ -30,6 +30,7 @@ def current_span
       def in_span(name, attributes: nil, links: nil, start_timestamp: nil, kind: nil, sampling_hint: nil)
         span = start_span(name, attributes: attributes, links: links, start_timestamp: start_timestamp, kind: kind, sampling_hint: sampling_hint)
         with_span(span) { |s| yield s }
+        span.finish
```

Comment:
This needs to capture the result of the `with_span` (technically, the result of its block) and return it after the `span.finish`. `.tap` may be appropriate here.

Reply:
@fbogsany I think I see a way to do that. Thanks for the behavior check!

Reply:
It just occurred to me that this should be in an `ensure`. I.e.:
```ruby
  with_span(span) { |s| yield s }
ensure
  span.finish
end
```
which has the added advantage of not requiring an extra local variable.